### PR TITLE
Allow custom logs to be added to frontail

### LINKED ISSUE #1445: Integration of HABApp logs in frontail
I just upgraded openHABian and reinstalled HABApp with the new openHABian option. Very smooth!
Since HABApp now logs to the default openHAB log directory, I think it would make sense to integrate better with frontail.

Please also see: https://github.com/spacemanspiff2007/HABApp/issues/202

I have done this in my own openHABian setup and think it would benefit other users if it would be done automatically by openhabian-config. A minor part was a change in HAPApp that @spacemanspiff2007 already have taken care of, the other part is in the openhabian setup.

The changes required are small:

1. In `/etc/systemd/system/frontail.service`:
    add `/var/log/openhab/HABApp.log` to `ExecStart=/usr/lib/node_modules/frontail/bin/frontail...`

2. In `/usr/lib/node_modules/frontail/preset/openhab.json`:
    add `"/var/log/openhab/HABApp.log": "text-align: right; font-size: 0.8em; border-top: 2px solid #F8F8F8;",` in the lines section. (Just realised that i forgot to update the `openhab_dark.json` as well!!)

I have tried to incorporate this into the openhabian installation scripts, but even if the function below works well,
I don't know how to handle the interdependency between frontail and HAPApp, since they can be installed and removed independently of each other. To have the code below in both `setup_frontail` and `setup_habapp` is quite messy and I think the function is too specialised to fit in `helpers.bash`.

I would greatly appreciate if I could get some guidance on where I should add this function so I can submit it as a PR.

```
## Function for integrating HABApp logs with frontail
## Valid arguments: "install" or "remove"
##
##    habapp_frontail_integration()
##
habapp_frontail_integration() {
  local frontailService="/etc/systemd/system/frontail.service"
  local frontailJSON="/usr/lib/node_modules/frontail/preset/openhab.json"
  local habappLogFile="/var/log/openhab/HABApp.log"
  local openhabLogFile="/var/log/openhab/openhab.log"

  if [[ $1 == "remove" ]]; then
    echo -n "$(timestamp) [openHABian] Removing HABApp-frontail log integration ... "
    if ! cond_redirect sed -i "/^ExecStart=/ s= ${habappLogFile}==" "${frontailService}"; then echo "FAILED (update frontail service file)"; return 1; fi
    if ! cond_redirect sed -i "\\=${habappLogFile}=d" "${frontailJSON}"; then echo "FAILED (update frontail preset file)"; return 1; fi
    if ! cond_redirect systemctl -q daemon-reload &> /dev/null; then echo "FAILED (daemon-reload)"; return 1; fi
    if cond_redirect systemctl restart frontail.service; then echo "OK"; else echo "FAILED (restart frontail)"; return 1; fi

    if [[ -n "$INTERACTIVE" ]]; then
      whiptail --title "HABApp-frontail log integration removed" --msgbox "We permanently removed the HABApp-frontail log integration from your system." 7 80
    fi
    return 0;
  fi
  if [[ $1 != "install" ]]; then return 1; fi

  echo -n "$(timestamp) [openHABian] Installing HABApp-frontail log integration ... "
  if [[ $( ! grep -q "^ExecStart=.*${habappLogFile}" "${frontailService}" ) ]]; then
    if ! cond_redirect sed -i "/^ExecStart=/ s=\$= ${habappLogFile}=" "${frontailService}"; then echo "FAILED (update frontail service file)"; return 1; fi
  fi
  if [[ $( ! grep -q "${habappLogFile}" "${frontailJSON}" ) ]]; then
    if ! cond_redirect sed -i "\\=${openhabLogFile}=p; s=${openhabLogFile}=${habappLogFile}=" "${frontailJSON}"; then echo "FAILED (update frontail preset file)"; return 1; fi
  fi
  if ! cond_redirect systemctl -q daemon-reload &> /dev/null; then echo "FAILED (daemon-reload)"; return 1; fi
  if cond_redirect systemctl restart frontail.service; then echo "OK"; else echo "FAILED (restart frontail)"; return 1; fi


  if [[ -n "$INTERACTIVE" ]]; then
    whiptail --title "HABApp-frontail log integration installed" --msgbox "HABApp-frontail log integration was successfully installed on your system." 7 80
  fi
  return 0;

}

```

### PR DESCRIPTION
**Pull request by ecdye**
Fixes #1445

Signed-off-by: Ethan Dye <mrtops03@gmail.com>

### CODE REVIEW

**Review by mstormi — APPROVED** [review]
LGTM, however I wonder if it's worth to add a "remove" function, too.
See use of select_blkdev() how to create a list in whiptail to choose from.

File: functions/nodejs-apps.bash [hunk 1]
```diff
@@ -105,40 +105,57 @@ frontail_setup() {
...
+    whiptail --title "Frontail not installed" --msgbox "Frontail is not installed!\\n\\nCanceling operation!" 9 80
+    return 0
+  fi
 
-    if [[ -f $newLog ]]; then
-      echo -n "$(timestamp) [openHABian] Adding '${newLog}' to frontail... "
-      if ! cond_redirect sed -i -e "/^ExecStart/s/$/ ${newLog}/" $frontailService; then echo "FAILED (add log)"; return 1; fi
-      if ! cond_redirect systemctl -q daemon-reload &> /dev/null; then echo "FAILED (daemon-reload)"; return 1; fi
-      if cond_redirect systemctl restart frontail.service; then echo "OK"; else echo "FAILED (restart service)"; return 1; fi
+  if [[ $1 == "add" ]]; then
```
Comment (mstormi) [inline functions/nodejs-apps.bash:130]: Does it work unattended ? Haven't seen it being called. What's the openhabian.conf parameter to set ?

Note that if not INTERACTIVE there will not be a $1 and if so this test is superfluous/unneeded

File: functions/nodejs-apps.bash [hunk 2]
```diff
@@ -105,40 +105,57 @@ frontail_setup() {
...
+          whiptail --title "File does not exist" --msgbox "The specifed file path does not exist!\\n\\nCanceling operation!" 9 80
+          return 0
+        else
+          echo "$(timestamp) [openHABian] Adding '${log}' to frontail... FAILED (file does not exist)"
+        fi
+      fi
+    done
+  elif [[ $1 == "remove" ]] && [[ -n $INTERACTIVE ]]; then
+    readarray -t array < <(grep -e "^ExecStart.*$" "$frontailService" | awk '{for (i=12; i<=NF; i++) {printf "%s\n\n", $i}}')
+    ((count=${#array[@]} + 6))
```
Comment (mstormi) [inline functions/nodejs-apps.bash:154]: please don't expect me to understand all the code
DNT but I believe you did

**Review by mstormi — CHANGES_REQUESTED** [review]

File: functions/nodejs-apps.bash (hunk 2 above)
Comment (ecdye) [reply functions/nodejs-apps.bash:154]: Sure did.

File: functions/nodejs-apps.bash (hunk 1 above)
Comment (ecdye) [reply functions/nodejs-apps.bash:130]: Nope, apparently I forgot to do that part and only changed the function itself, I'll fix that.

File: build-image/openhabian.conf [hunk 3]
```diff
@@ -89,4 +89,7 @@ storagecapacity=1024
 # Tailscale VPN
 #preauthkey=tskey-xxxxxxxxxxxxxxxxx
 
+# Custom log files
+#custom_log_files=("/var/log/foo.log" "/var/log/bar.log")
```
Comment (mstormi) [inline build-image/openhabian.conf:93]: I guess that format will confuse users. I don't mind but would prefer "file1 file2"

**Review by mstormi — APPROVED** [review]

File: build-image/openhabian.conf (hunk 3 above)
Comment (ecdye) [reply build-image/openhabian.conf:93]: Oh well. Users who don't understand that format probably don't know where or what the log files will be ahead of time anyway.

**Pull request merged**